### PR TITLE
Added command: `nf-core modules remove`

### PR DESCRIPTION
**Pull request by KevinMenden**
## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

This PR adds the command `nf-core modules remove` which allows to uninstall/remove modules in a pipeline repository. This currently only works for modules from `nf-core/modules`, because for all other modules in the `modules/local` directory there isn't any structure.

### CODE REVIEW

File: nf_core/modules.py [hunk 1]
```diff
@@ -101,7 +95,33 @@ def update(self, module, force=False):
...
+        # Remove the module
+        try:
+            shutil.rmtree(module_dir)
+            log.info("Successfully removed {} module".format(module))
+            return True
+        except OSError as e:
+            log.error("Could not remove module: {}".format(e))
+            return False
+
         pass
```
Comment (ewels) [inline nf_core/modules.py:125]:
```suggestion
```
This was only needed as it was an empty function before 👍🏻

File: nf_core/modules.py [hunk 2]
```diff
@@ -202,3 +222,14 @@ def download_gh_file(self, dl_filename, api_url):
         # Write the file contents
         with open(dl_filename, "wb") as fh:
             fh.write(file_contents)
+
+    def has_valid_pipeline(self):
+        # Check that we were given a pipeline
```
Comment (ewels) [inline nf_core/modules.py:227]:
```suggestion
        """Check that we were given a pipeline
        """
```

File: tests/test_modules.py [hunk 3]
```diff
@@ -58,3 +58,12 @@ def test_modules_install_fastqc_twice(self):
         """ Test installing a module - FastQC already there """
         self.mods.install("fastqc")
         assert self.mods.install("fastqc") is False
+
+    def test_modules_remove_fastqc(self):
+        """ Test removing FastQC module after installing it"""
+        self.mods.install("fastqc")
+        assert self.mods.remove("fastqc")
```
Comment (ewels) [inline tests/test_modules.py:65]: Might be nice to assert that the files are actually gone too 😄

**Review by ewels — APPROVED** [review]
Great! 🚀 

Needs documentation in the readme. Could do with a bit of a docs overhaul generally as it's getting very long, but that's for another time.

File: tests/test_modules.py (hunk 3 above)
Comment (KevinMenden) [reply tests/test_modules.py:65]: 👀 
Good point right there 😄

**Comment by KevinMenden** [comment]
Thanks for the review @ewels !
I put adding docs on my todo-list for this week, wanted to add a few more commands now anyway. Wasn't aware that there is not documentation for `nf-core modules` at all currently, at least not in the README.

**Pull request merged**